### ISSUE TITLE
MarkovChain solver should throw error on double cycle

### ISSUE THREAD

**Issue by DrPaulBrewer** [DrPaulBrewer/gridvoting#29]
test case:

6 items, two separate 3-item cycles with no overlap

```
P = np.array([
    [1/2,1/2,0,0,0,0],
    [0,1/2,1/2,0,0,0],
    [1/2,0,1/2,0,0,0],
    [0,0,0,1/2,1/2,0],
    [0,0,0,0,1/2,1/2],
    [0,0,0,1/2,0,1/2]
])
```

Desired outcome:

Since stationary distribution is not unique, throw an error.  

Current outcome:

`.find_unique_stationary_distribution()` loops forever

`.solve_for_unit_eigenvector()` will throw `LinAlgError: Singular matrix` on CPU and will either return `[nan,...,nan]` or throw an error on GPU.   The errors thrown are not always obvious.

**Comment by DrPaulBrewer** [comment]
This seems fixed in commit 10cdecc

Passed on CPU.
Passed on A100-40GB GPU.

**Comment by DrPaulBrewer** [comment]
Closed by PR #28  v0.70
